File: src/common/web/services/ServiceContext.ts
import { MessageContext } from "../core/messaging/handlers/MessageContext";

/**
 * An execution context for messages dispatched by the message bus to a service.
 *
 * This class is an extension to the more general ``MessageContext`` specifically used by ``Service`` and its message handlers.
 */
export class ServiceContext extends MessageContext {
}

File: src/common/py/services/service_context.py
import typing

from ..core.messaging.handlers import MessageContext


class ServiceContext(MessageContext):
    """
    An execution context for messages dispatched by the message bus to a service.

    This class is an extension to the more general ``MessageContext`` specifically used by ``Service`` and its message handlers.
    """


ServiceContextType = typing.TypeVar(
    "ServiceContextType", bound=ServiceContext
)  # pylint: disable=invalid-name
